AEPTarget/Sources/Target.swift

```diff
@@ -536,6 +619,66 @@ public class Target: NSObject, Extension {
         }
     }
 
+    private func processTargetRawRequestResponse(event: Event, connection: HttpConnection) {
+        var error: String? = (connection.responseMessage != TargetConstants.NetworkConnection.CONNECTION_RESPONSE_MESSAGE_NO_ERROR) ? connection.responseMessage : nil
+        var responseData: [[String: Any]]?
+
+        defer {
+            dispatchExecuteRawResponse(event: event, error: error, data: responseData)
+        }
+
+        guard
+            let data = connection.data,
+            let responseDictAnyCodable = try? JSONDecoder().decode([String: AnyCodable].self, from: data),
+            let responseDict = AnyCodable.toAnyDictionary(dictionary: responseDictAnyCodable)
+        else {
+            Log.debug(label: Target.LOG_TAG, "Target response parser initialization failed")
+            error = TargetError.ERROR_RESPONSE_PARSING_FAILED
+            return
+        }
+
+        let response = TargetDeliveryResponse(responseJson: responseDict)
+
+        if connection.responseCode != 200 {
+            if let responseError = response.errorMessage {
+                if responseError.contains(TargetError.ERROR_NOTIFICATION_TAG) {
+                    targetState.clearNotifications()
+                }
+                error = responseError
+            } else {
+                error = connection.error?.localizedDescription ?? error
+            }
+            Log.debug(label: Target.LOG_TAG, "Target raw execute request failed with response code: \(String(describing: connection.responseCode)) and error: \(String(describing: error)))")
+            return
+        }
+
+        // Clear the notifications when the response is 200 OK
+        targetState.clearNotifications()
+
+        if let tntId = response.tntId { targetState.updateTntId(tntId) }
+        if let edgeHost = response.edgeHost { targetState.updateEdgeHost(edgeHost) }
+        createSharedState(data: targetState.generateSharedState(), event: nil)
```

Comment:
why not pass `event` here?

Reply:
Good catch! Fixed it in the existing method, for processing retrieve location content response, as well.